lnwallet/channel.go

```diff
@@ -5532,6 +5532,39 @@ func (lc *LightningChannel) NumLocalUpdatesPendingOnRemote() uint64 {
 	return lc.localUpdateLog.logIndex - lastRemoteCommit.ourMessageIndex
 }
 
+// NumLocalUpdatesPendingOnLocal returns the number of local updates that still
+// need to be applied to the local commitment tx.
+func (lc *LightningChannel) NumLocalUpdatesPendingOnLocal() uint64 {
+	lc.RLock()
+	defer lc.RUnlock()
+
+	lastLocalCommit := lc.localCommitChain.tip()
+
+	return lc.localUpdateLog.logIndex - lastLocalCommit.ourMessageIndex
+}
+
+// NumRemoteUpdatesPendingOnLocal returns the number of remote updates that
+// still need to be applied to the local commitment tx.
+func (lc *LightningChannel) NumRemoteUpdatesPendingOnLocal() uint64 {
+	lc.RLock()
+	defer lc.RUnlock()
+
+	lastLocalCommit := lc.localCommitChain.tip()
+
+	return lc.remoteUpdateLog.logIndex - lastLocalCommit.theirMessageIndex
+}
+
+// NumRemoteUpdatesPendingOnRemote returns the number of remote updates that
+// still need to be applied to the remote commitment tx.
+func (lc *LightningChannel) NumRemoteUpdatesPendingOnRemote() uint64 {
+	lc.RLock()
+	defer lc.RUnlock()
+
+	lastRemoteCommit := lc.remoteCommitChain.tip()
+
+	return lc.remoteUpdateLog.logIndex - lastRemoteCommit.theirMessageIndex
```

Comment:
Seems possible to overflow since `remoteUpdateLog.logIndex` could be less than the latter?

Reply:
If this is ever negative we have serious problems.